src/clients/github.js

```diff
@@ -117,54 +99,89 @@ export function createGitIgnore(projectName) {
   }
 }
 
+
 /**
  * Initialize an empty repository within a git repo, add the gitignore, and the rest of the files.
  * Commit them, create the remote repository and push the commit to the remote repository.
- * @param configs
- * @param githubToken
- * @param githubCredentials
+ *
+ * @param projectName
+ * @param projectDescription
+ * @param username
+ * @param password
  */
-export function createGitHubRepository(configs, githubToken, githubCredentials) {
-  winston.log('verbose', 'createGitHubRepository', { projectName: configs.projectName, githubToken });
-
-  const data = {
-    name: configs.projectName,
-    description: configs.description,
-    private: false
-  };
-
-  github.authenticate({
-    type: 'oauth',
-    token: githubToken
+export function createGitHubRepository(projectName, projectDescription, username, password) {
+  winston.log('verbose', 'createGitHubRepository', { projectName });
+
+  return new Promise((resolve, reject) => {
+    superagent
+      .post(`${githubApiUrl}/user/repos`)
+      .set({
+        Authorization: basicAuthorization(username, password)
+      })
+      .send({
+        name: projectName,
+        description: projectDescription,
+        private: false
+      })
+      .end((err) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve();
+        }
+      });
+  });
+}
+
+
+/**
+ * Init the git repository, add all the files, make the first commit,
+ * add the remote origin, and push origin to master.
+ *
+ * @param username
+ * @param projectName
+ */
+export function initAddCommitAndPush(username, projectName) {
+  winston.log('verbose', 'initAddCommitAndPush', { username, projectName });
+  winston.log('info', 'Pushing all files to the new git repository...');
+
+  return new Promise((resolve) => {
+    git(`${process.cwd()}/${projectName}`)
+      .init()
+      .add('.gitignore')
+      .add('./*')
+      .commit('Initial commit')
+      .addRemote('origin', `https://github.com/${username}/${projectName}.git`)
+      .push('origin', 'master')
+      .exec(() => {
+        console.log('Please wait while the files are uploaded...');
+        setTimeout(() => {
+          resolve();
+        }, 10000); // TODO: Find a better way to do this than a timeout
```

Comment:
Why do we have this timeout? You could use this for a cleaner option - https://github.com/Secbone/es7-sleep

Reply:
I agree, we should find a better solution than a timeout. But maybe we can open another issue for it. This PR is already huge!

Reply:
👍